src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -135,16 +135,27 @@ public override unsafe int FindFirstCharacterToEncode(char* text, int textLength
                     // Load the next 16 characters, combine them to one byte vector.
                     // Chars that don't cleanly convert to ASCII bytes will get converted (saturated) to
                     // somewhere in the range [0x7F, 0xFF], which the NeedsEscaping method will detect.
-                    Vector128<sbyte> sourceValue = Sse2.PackSignedSaturate(
-                        Sse2.LoadVector128(ptr),
-                        Sse2.LoadVector128(ptr + Vector128<short>.Count));
+                    Vector128<sbyte> sourceValue;
+
+                    if (Sse2.IsSupported)
+                    {
+                        sourceValue = Sse2.PackSignedSaturate(
+                            Sse2.LoadVector128(ptr),
+                            Sse2.LoadVector128(ptr + Vector128<short>.Count));
+                    }
+                    else
+                    {
+                        Debug.Assert(AdvSimd.Arm64.IsSupported);
```

Comment:
>Debug.Assert(AdvSimd.Arm64.IsSupported); [](start = 24, length = 40)

Just for safety in future, I would update this to `else if (AdvSimd.Arm64.IsSupported)` because that is optimized by JIT.

Reply:
My intention here is to guard against potential future architecture additions. Assuming somebody adds a check for a new architecture at the top of the method but forgets to include a check here, it would fail fast with an assert failure. Changing this to `else if (AdvSimd.Arm64.IsSupported)` would make it more difficult to diagnose.

Reply:
It would likely be good to have it be `else if (...) { } else { throw PlatformNotSupportedException(); }`